### PR TITLE
fix: Add translation hooks on settings screen

### LINKED ISSUE #2117: Add translation hooks where missing
There are quite a few spots still where English strings are hardcoded, these need moving to use `react-i18n` and (ideally but not required) functional components. A lot of the remaining spots are in confirmation dialogs, modals, toast messages etc.

Fixing several screens at a time and doing multiple smaller PR's seems like the best way to get these tackled in a manageable manner. These are the components I'm currently aware:

- [x] `ErrorBoundary`
- [x] `InputRich`
- [x] `CollectionDelete`
- [x] `CollectionExport`
- [x] `DocumentTemplatize`
- [x] `DocumentMove`
- [x] `DocumentNew`
- [x] `GroupDelete`
- [x] `GroupEdit`
- [x] `GroupNew`
- [x] `GroupMembers`
- [x] `Invite`
- [x] `UserDelete` 
- [x] `Settings/Details`
- [x] `Settings/Notifications`
- [x] `Settings/Security`
- [x] `Settings/Tokens`
- [x] `Settings/Zapier`

### PR DESCRIPTION
**Pull request by iamsaumya**
towards #2117 

One of many to come.

### CODE REVIEW

**Comment by iamsaumya** [comment]
I was confused about whether to use `<Trans>` or not.

File: app/scenes/Settings/Details.js [hunk 1]
```diff
@@ -14,137 +13,130 @@ import Input, { LabelText } from "components/Input";
...
 import env from "env";
-
-type Props = {
-  auth: AuthStore,
-  ui: UiStore,
-};
-
-@observer
-class Details extends React.Component<Props> {
-  timeout: TimeoutID;
```
Comment (iamsaumya) [inline app/scenes/Settings/Details.js:25]: I couldn't find `timeout` usage

File: app/scenes/Settings/Details.js [hunk 2]
```diff
@@ -14,137 +13,130 @@ import Input, { LabelText } from "components/Input";
...
-    return (
-      <Scene title="Details" icon={<TeamIcon color="currentColor" />}>
-        <Heading>Details</Heading>
-        <HelpText>
+import useCurrentTeam from "hooks/useCurrentTeam";
+import useStores from "hooks/useStores";
+
+function Details() {
+  const { auth, ui } = useStores();
+  const team = useCurrentTeam();
```
Comment (iamsaumya) [inline app/scenes/Settings/Details.js:21]: Do we need to use the `useCurrentTeam` approach to access the team rather than `auth.team`?

Comment (tommoor) [reply app/scenes/Settings/Details.js:21]: It's preferred 👍

File: app/scenes/Settings/Details.js [hunk 3]
```diff
@@ -14,137 +13,130 @@ import Input, { LabelText } from "components/Input";
...
+  const handleAvatarError = React.useCallback(
+    (error: ?string) => {
+      ui.showToast(error || t("Unable to upload new logo"));
+    },
+    [ui, t]
+  );
+
+  const isValid = form.current && form.current.checkValidity();
+
+  if (!team) return null;
```
Comment (tommoor) [inline app/scenes/Settings/Details.js:76]:
```suggestion
```

Shouldn't be required if using `useCurrentTeam`

File: app/scenes/Settings/Details.js [hunk 4]
```diff
@@ -14,137 +13,130 @@ import Input, { LabelText } from "components/Input";
...
+            {subdomain && (
+              <HelpText small>
+                <Trans>Your knowledge base will be accessible at</Trans>{" "}
+                <strong>{subdomain}.getoutline.com</strong>
+              </HelpText>
+            )}
+          </>
+        )}
+        <Button type="submit" disabled={auth.isSaving || !isValid}>
+          {auth.isSaving ? t("Saving…") : t("Save")}
```
Comment (tommoor) [inline app/scenes/Settings/Details.js:135]:
```suggestion
          {auth.isSaving ? `${t("Saving")}…` : t("Save")}
```

It's a pita from a syntax pov but it's better for the translations

File: app/scenes/Settings/Notifications.js [hunk 5]
```diff
@@ -1,137 +1,141 @@
...
+      }
+
+      showSuccessMessage();
+    },
+    [notificationSettings, showSuccessMessage]
+  );
+
+  const showSuccessNotice = window.location.search === "?success";
+
+  if (!team || !user) return null;
```
Comment (tommoor) [inline app/scenes/Settings/Notifications.js:88]: Not required if we change to `useCurrentUser` and `useCurrentTeam` hooks

**Comment by tommoor** [comment]
> I was confused about whether to use <Trans> or not.

As a general rule I use it for large blocks of text (it looks better), or places where it's required due to react tags mixed in with the translation.

**Review by tommoor — APPROVED** [review]

**Pull request merged**